### DISCUSSION TITLE
How to add I2C peripheral in the Bare Metal AMP code?

### DISCUSSION THREAD

**Discussion by chu551122** [polarfire-soc/polarfire-soc-documentation#185] [Applications and demos] [answered]
Hi, Sir:
   I want to add I2C in the Bare-metal AMP example code, I met the error message even I think I've added the files right as the screenshot, is there anything I missed or need to set so as to solve the compiling error?
  Thank you very much! 
![image](https://user-images.githubusercontent.com/55179520/186123239-a7a994e2-58df-4ccd-a0dd-48e70d1529be.png)
![image](https://user-images.githubusercontent.com/55179520/186124556-b42b5654-c926-4e3a-bedc-65dc320439b7.png)
![image](https://user-images.githubusercontent.com/55179520/186125091-bdb63cc0-d77f-4da8-9872-c88e241597aa.png)

**Comment by hughbreslin** [comment, marked as answer]
HI @chu551122 It looks like you checked this out using Git, can you make sure that all of the resources are in the same source directory? 

Select (for example) the "demo_main.c" file and right click and select "properties" - this should give you the full path to the file, then do the same for your mcp9902 driver and see if it all lines up. 

Also have you updated the make files for this project to include the additional sources you've added?

**Reply by chu551122** [reply]
Dear Hugh:
   Yes, you are right, after I modified the Makefile, it's compiled ok.
   Thank you very much!
![image](https://user-images.githubusercontent.com/55179520/186286670-505c3f97-494f-4256-a600-57bf9638b899.png)
